app/V1Module/security/ACL/ISubmissionFailurePermissions.php

```diff
@@ -2,12 +2,12 @@
 namespace App\Security\ACL;
 
 
-use App\Model\Entity\Submission;
+use App\Model\Entity\AssignmentSolutionSubmission;
 use App\Model\Entity\SubmissionFailure;
 
 interface ISubmissionFailurePermissions {
   function canViewAll(): bool;
   function canView(SubmissionFailure $failure): bool;
   function canResolve(SubmissionFailure $failure): bool;
-  function canViewForSubmission(Submission $submission): bool;
-}
\ No newline at end of file
+  function canViewForSubmission(AssignmentSolutionSubmission $submission): bool;
```

Comment:
I would probably rename this method to canViewForAssignment[Solution]Submission

Reply:
Will do